backend/devices/bitbox02/keystore.go

```diff
@@ -452,14 +453,65 @@ func (keystore *keystore) signBTCTransaction(btcProposedTx *btc.ProposedTransact
 	if btcProposedTx.FormatUnit == coinpkg.BtcUnitSats {
 		formatUnit = messages.BTCSignInitRequest_SAT
 	}
+
+	// Payment request handling
+	newBTCPaymentRequest := func(txPaymentRequest *accounts.PaymentRequest) (*messages.BTCPaymentRequestRequest, error) {
+		memos := []*messages.BTCPaymentRequestRequest_Memo{}
+		for _, m := range txPaymentRequest.Memos {
+			memo := messages.BTCPaymentRequestRequest_Memo{
+				Memo: &messages.BTCPaymentRequestRequest_Memo_TextMemo_{
+					TextMemo: &messages.BTCPaymentRequestRequest_Memo_TextMemo{
+						Note: m.Note,
+					},
+				},
+			}
+			memos = append(memos, &memo)
+		}
+
+		sigBytes, err := hex.DecodeString(txPaymentRequest.Signature)
+		if err != nil {
+			return nil, err
+		}
+
+		nonceBytes, err := hex.DecodeString(txPaymentRequest.Nonce)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
It's not clear the request-address nonce is supposed to be hex-encoded. In my above comment I suggest to just error if it is not empty and use `Nonce []byte`, so you can skip the conversion here.

@davidknezic could you make the nonce type `Uint8Array` for clarity here? Or remove it and re-add it only once we needed.

https://github.com/pocketbitcoin/request-address/blob/main/src/slip24.ts#L5

Reply:
Also, could it make sense that the payment request message contains a custom fee that Pocket may want to suggest for the transaction?

Reply:
Not sure. It's not in SLIP24, and to me it feels like the sender should have control over the fee, not the recipient.

Reply:
@thisconnect the sender in this case is the BitBoxApp, not the user :P It chooses the highest fee rate for now.